OpenRA.Mods.Common/Traits/World/WeatherOverlay.cs

```diff
@@ -0,0 +1,527 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Linq;
+using OpenRA.Graphics;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Adds an overlay for weather animations based on dropping particles like snowfalkes, raindrops or sand grains.")]
+	public class WeatherOverlayInfo : ITraitInfo
+	{
+		[Desc("Factor for particle density. As smaller the value as more partikles will get spawned.")]
+		public readonly int ParticleDensityFactor = 1311;
+
+		[Desc("Sets the kind of weather particles")]
+		public readonly WeatherOverlay.WeatherVariants  WeatherVariants =WeatherOverlay.WeatherVariants.snow;
+
+		[Desc("Makes dropping particles not falling straight down.")]
+		public readonly bool Wind = true;
+
+		[Desc("The wind is fading in from zero.")]
+		public readonly bool InstantWind = false;
+
+		[Desc("The wind will change in intensity and direction over time.")]
+		public readonly bool ChangingWind = true;
+
+		[Desc("This works only when ChangingWind is disabled. It sets the wind intensity and direction (sign) manually. Values between -5 and 5 are used when ChangingWind is enabled.")]
+		public readonly int WindStrength;
+		
+		[Desc("The first color of the snow particles. The default is the 128th. Color from the temperate.pal")]
```

Comment:
Can we get an array of Color instead? That way we will only have one color field.